Apps/WebClient/src/ClientApp/src/components/calendar/body.vue

```diff
@@ -201,17 +204,33 @@ export default class CalendarBodyComponent extends Vue {
             };
 
             for (let perDay = 0; perDay < 7; perDay++) {
-                week.days.push({
+                let dayEvent = {
                     id: startDate.getTime().toString() + "-" + perDay,
                     monthDay: startDate.getDate(),
                     isToday: now.toDateString() == startDate.toDateString(),
                     isCurMonth: startDate.getMonth() == current.getMonth(),
                     weekDay: perDay,
                     date: new Date(startDate),
                     events: this.slotEvents(startDate),
-                });
+                };
+                week.days.push(dayEvent);
 
                 startDate.setDate(startDate.getDate() + 1);
+
+                if (
+                    this.firstEventDateOfCurrentMonth === null &&
```

Comment:
I dont think this is necessary. I would use a lambda instead to find the first event day that is on this month, then the first day from that

Reply:
I'm not sure how it can be better, but I will try. Btw, it is not an easy change as this logic needs to be tested carefully as well

Reply:
Per our discussion, created 8741 for this refactoring.